Resources/public/js/views/services/plugins/ez-locationsearchplugin.js

```diff
@@ -96,6 +125,121 @@ YUI.add('ez-locationsearchplugin', function (Y) {
+                            return;
+                        }
+                    });
+
+                this._loadContentForLocations(locationStructArr, endContentLoad);
+            }
+
+            tasks.done(function () {
+                callback(loadResourcesError, locationStructArr);
+            });
+        },
+
+        /**
+         * Loads ContentType for each location struct in given array and puts it in the new `contentType`
+         * field for the location struct.
+         *
+         * @method _loadContentTypeForLocations
+         * @protected
+         * @param {Array|Null} locationStructArr
+         * @param {Function} callback
+         */
+        _loadContentTypeForLocations: function (locationStructArr, callback) {
+            var tasks = new Y.Parallel(),
+                loadContentTypeError = false;
+
+            Y.Array.each(locationStructArr, function (locationStruct, i) {
+                var contentType = new Y.eZ.ContentType(
+                        {id: locationStruct.location.get('contentInfo').get('resources').ContentType}
+                    ),
+                    capi = this.get('host').get('capi'),
+                    options = {api: capi},
+                    endContentTypeLoad = tasks.add(function (error, response) {
+                        if (error) {
+                            loadContentTypeError = true;
+                            return;
+                        }
+
+                        locationStructArr[i].contentType = contentType;
+                    });
+
+                contentType.load(options, endContentTypeLoad);
+            }, this);
+
+            tasks.done(function () {
+                callback(loadContentTypeError, locationStructArr);
+            });
+        },
+
+        /**
+         * Loads Content for each location struct in given array and puts it in the new `content`
+         * field for the location struct.
+         *
+         * @method _loadContentForLocations
+         * @protected
+         * @param {Array|Null} locationStructArr
+         * @param {Function} callback
+         */
+        _loadContentForLocations: function (locationStructArr, callback) {
+            var tasks = new Y.Parallel(),
+                loadContentError = false;
+
+            Y.Array.each(locationStructArr, function (locationStruct, i) {
+                var content = new Y.eZ.Content(
+                        {id: locationStruct.location.get('contentInfo').get('id')}
+                    ),
+                    capi = this.get('host').get('capi'),
+                    options = {api: capi},
+                    endContentLoad = tasks.add(function (error, response) {
+                        if (error) {
+                            loadContentError = true;
+                            return;
+                        }
+
+                        locationStructArr[i].content = content;
+                    });
+
+                content.load(options, endContentLoad);
```

Comment:
instead of loading each content individually, we can load them in one request by doing a content search like it's done in [the content tree plugin](https://github.com/ezsystems/PlatformUIBundle/blob/master/Resources/public/js/views/services/plugins/ez-contenttreeplugin.js#L140). It's too late to do that right now though. Can you create a follow up improvement request in JIRA for that ?